test/blackbox_test.proto

```diff
@@ -0,0 +1,129 @@
+// BlackBoxTest represents a list of test APIs that facilitates testing the
+// switch/network (injecting events, mutating switch state etc.)
+
+syntax = "proto3";
+
+package gnoi.test;
+
+//import "enums/enums.proto";
+import "google/rpc/status.proto";
+import "types/types.proto";
+
+
+service BlackBoxTest {
+  // This RPC allows the state of a transceiver (for both optical, and copper
+  // cable) to be set on the switch. The goal is to simulate the
+  // insertion/removal of a transceiver and verify that the switch initializes
+  // the transceiver correctly. This RPC maps to the `fake` removal of a
+  // physical module, and un-doing the `fake` removal (i.e., insert). This does
+  // not implement any fake insertion of a non-present module. The switch
+  // remembers the state of a transceiver based on this RPC (until reboot).
+  rpc SetTransceiverState(SetTransceiverStateRequest)
+      returns (SetTransceiverStateResponse) {}
+
+  // This RPC changes the state of the link connected to the specified port(s).
+  // The goal is to simulate a link going up/down due to factors external to the
+  // switch: copper cable/fiber removed, cable/fiber gone bad, etc.
+  rpc SetHardwareLinkState(SetHardwareLinkStateRequest)
```

Comment:
I think setting link state is also a configuration, not a gNOI?

If the goal is to simulate 'loss of light' at an optical level, then we could add a leaf to the [transceiver tree](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-platform.html#components-component-transceiver-physical-channels-channel-config)
something like 
`/openconfig-platform:components/component/openconfig-platform-transceiver:transceiver/physical-channels/channel/config/rx-laser-softdisable`. When set to true, the laser receiver would be 'soft disabled', causing that's physical channel to be simulated as down.   

Alternatively, this could also be done at higher layers 2 or 3 via /interfaces.  This might be more simple and cover more use cases (being neutral to things like copper/optical and physical channels). But I am not sure if your use case requires L1 optical layer simulation.